src/utils/Helper.ts

```diff
@@ -43,3 +44,15 @@ export function applySomeDefaults<T>(defaults: Partial<T>, partial?: Partial<T>)
   const complete = Object.assign(defaults, partial)
   return complete
 }
+
+export function mergeRequestOptions<T = CompleteRequestOptions | RequestOptions>(
+  aRequestOption: T,
+  anotherRequestOption: RequestOptions
+): T {
+  const headers: Record<string, string> = {
+    ...(aRequestOption as RequestOptions).headers,
```

Comment:
Just to confirm, if `aRequestOption.headers` is not set, does this work or fail?

Reply:
It works, I've just added a test for that case :)